### DISCUSSION TITLE
Am I able to use rustube in a non-synchronous way?

### DISCUSSION THREAD

**Discussion by nednoodlehead** [DzenanJupic/rustube#50] [Q&A] [answered]
I see lots of the docs.rs documentation uses tokio / async / await functions, I've never really used any of that stuff before, does this crate have support for non-async usage? I'm basically hoping to just have `pytube` in Rust that I can work with.

I saw [this part](https://docs.rs/rustube/latest/rustube/blocking/index.html) of the documentation that mentions it being a "Blocking wrappers for using rustube in a synchronous context.". But running that code at the top of the link just spits out an error (? operator can only be applies to values that implement "try") on the `Video::from_url(&url)?` part.

I would really assume that this "problem" is just user error, but I've read through a fair bit of docs and haven't found an example that runs without the tokio async stuff.

Any help / suggestions would be appreciated

**Comment by DzenanJupic** [comment, marked as answer]
Hey @nednoodlehead, yes `rustube` provides blocking wrappers. These wrappers have the same name as the non-blocking types. So my guess, without seeing any of the code, is that you're using `rustube::Video` instead of `rustube::blocking::Video`. Also, you need to enable the `blocking` feature of the crate.

I hope this solves your issues. If not, please share the exact code you used.

**Reply by nednoodlehead** [reply]
Yes, this does solve things. Thanks.
